subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppLibraryPlugin.java

```diff
@@ -110,8 +114,7 @@ public void apply(final ProjectInternal project) {
         project.afterEvaluate(new Action<Project>() {
             @Override
             public void execute(final Project project) {
-                library.getTargetMachines().finalizeValue();
-                Set<TargetMachine> targetMachines = library.getTargetMachines().get();
+                Set<TargetMachine> targetMachines = setDefaultAndGetTargetMachineValues(library.getTargetMachines(), targetMachineFactory);
```

Comment:
Should probably also move the validation on the next 3 lines into the "setDefaultAndGetTargetMachineValues()` method as all of the callers do the same validation.